src/ssz/type/bit_list.zig

```diff
@@ -79,6 +79,41 @@ pub fn BitList(comptime limit: comptime_int) type {
             return true_bit_count;
         }
 
+        pub fn getSingleTrueBit(self: *const @This()) ?usize {
+            var found_index: ?usize = null;
+            const full_byte_len = self.bit_len / 8;
+            const remainder_bits = self.bit_len % 8;
+
+            for (0..full_byte_len) |i_byte| {
+                var b = self.data.items[i_byte];
+                while (b != 0) {
+                    if (found_index != null) {
+                        return null; // more than one true bit found
+                    }
+                    const lsb: usize = @as(u8, @ctz(b));
+                    const bit_index = i_byte * 8 + lsb;
+                    found_index = bit_index;
+
+                    b &= b - 1;
+                }
+            }
+            if (remainder_bits > 0) {
+                const tail_mask: u8 = (@as(u8, 1) << @intCast(remainder_bits)) - 1;
+                var b = self.data.items[full_byte_len] & tail_mask;
```

Comment:
what's this tail mask for? The implementation can be the same-ish here as for bitvector.
In this representation, there is no added padding bit (that is only added during serialization). In serialization, we also assume that all trailing bits are 0.

Reply:
ah yeah, imo we should fix resize to wipe trailing bits and trailing bytes (like how it zeros newly initialized bytes)
(In practice, I don't think we ever resize going bigger to smaller tho)